arena.go

```diff
@@ -0,0 +1,39 @@
+package bigqueue
+
+import (
+	"os"
+	"syscall"
+
+	"github.com/grandecola/mmap"
+)
+
+const (
+	cFilePerm = 0744
+)
+
+// Arena is an abstraction for a memory mapped file of a given size
+type Arena struct {
+	*mmap.Mmap
+	size int
+	file string
+}
+
+// NewArena returns pointer to an Arena. It takes a file location and mmaps it.
+// If file location does not exist, it creates file of given size.
+func NewArena(file string, size int) (*Arena, error) {
+	f, err := os.OpenFile(file, os.O_CREATE|os.O_RDWR, cFilePerm)
```

Comment:
Is it creating file with given size, as the function definition says? If yes, NewQueueIndex does not need to check for existence of file, it can directly call this function with file and size. If no, NewSharedFileMmap will throw an error as the region does not exist.